sdk/core/System.Net.ClientModel/src/Options/RequestOptions.cs

```diff
@@ -11,49 +11,92 @@ namespace System.Net.ClientModel;
 /// the message being sent down the pipeline.  For the duration of pipeline.Send,
 /// this may change some behaviors in various pipeline policies and the transport.
 /// </summary>
-// TODO: Make options freezable
-// Note: I was calling this RequestOptions, but I'm changing it back to RequestOptions.
-public class RequestOptions : PipelineOptions
+public class RequestOptions
 {
-    public virtual void Apply(ClientMessage message)
+    public RequestOptions()
+    {
+        PipelineOptions = new PipelineOptions();
+        ErrorBehavior = ErrorBehavior.Default;
+        CancellationToken = CancellationToken.None;
+    }
+
+    public RequestOptions(PipelineOptions pipelineOptions)
+    {
+        PipelineOptions = new(pipelineOptions);
+        ErrorBehavior = ErrorBehavior.Default;
+        CancellationToken = CancellationToken.None;
+    }
+
+    protected internal virtual void Apply(ClientMessage message)
     {
         // Wire up options on message
         message.CancellationToken = CancellationToken;
-        message.MessageClassifier = MessageClassifier ?? MessageClassifier.Default;
+        message.MessageClassifier = GetMessageClassifier();
 
         // TODO: note that this is a lot of *ways* to set values on the
         // message, policy, etc.  Let's get clear on how many ways we need and why
         // and when we use what, etc., then simplify it back to that per reasons.
-        if (NetworkTimeout.HasValue)
+        if (PipelineOptions.NetworkTimeout.HasValue)
         {
-            ResponseBufferingPolicy.SetNetworkTimeout(message, NetworkTimeout.Value);
+            ResponseBufferingPolicy.SetNetworkTimeout(message, PipelineOptions.NetworkTimeout.Value);
         }
     }
 
-    public virtual ErrorBehavior ErrorBehavior { get; set; } = ErrorBehavior.Default;
+    // Hard-codes precedence rules for MessageClassifier
+    private MessageClassifier GetMessageClassifier()
+    {
+        // TODO: We have a bug in this logic currently because classifiers are not chaining
+        if (MessageClassifier is not null)
+        {
+            return MessageClassifier;
+        }
+
+        if (PipelineOptions.MessageClassifier is not null)
+        {
+            return PipelineOptions.MessageClassifier;
+        }
+
+        return MessageClassifier.Default;
+    }
 
-    // TODO: handle duplication across message and options
+    public PipelineOptions PipelineOptions { get; }
 
-    public virtual CancellationToken CancellationToken { get; set; } = DefaultCancellationToken;
+    public virtual ErrorBehavior ErrorBehavior { get; set; }
 
-    #region Transport options - TODO: move to a subtype type?
+    public virtual CancellationToken CancellationToken { get; set; }
 
-    // TODO: Can we throw if someone gives us Transport options and transport isn't
-    // in the pipeline?
+    public virtual MessageClassifier? MessageClassifier { get; set; }
```

Comment:
MessageClassifier appears on both PipelineOptions and RequestOptions, and precedence rules are built-into RequestOptions.